Azure/azure-cli
# [AKS] Fix #25530: `az aks nodepool upgrade`: Fix agent pool property name used for fetching current k8s version

**Related command**

```sh
az aks nodepool upgrade
```

**Description**

Starting from version 2.45.0, the `az aks nodepool upgrade` version fails when used with the `--kubernetes-version` parameter. This was because of accessing a non-existent property on the `AgentPool` instance object here and here, which has been fixed in this PR.

resolves #25530 

**Testing Guide**

```sh
az aks nodepool upgrade --cluster-name {} --resource-group {} --nodepool-name {} --kubernetes-version {}
```

---

This checklist is used to make sure that common guidelines for a pull request are followed.
